operator/pkg/gateway-api/cell.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package gateway_api
+
+import (
+	"fmt"
+
+	"github.com/bombsimon/logrusr/v4"
+	"github.com/spf13/pflag"
+	ctrlRuntime "sigs.k8s.io/controller-runtime"
+
+	operatorOption "github.com/cilium/cilium/operator/option"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/logging"
+)
+
+// Cell manages the Gateway API related controllers.
+var Cell = cell.Module(
+	"gateway-api",
+	"Manages the Gateway API controllers",
+
+	cell.Config(gatewayApiConfig{
+		EnableGatewayAPISecretsSync: true,
+		GatewayAPISecretsNamespace:  "cilium-secrets",
+	}),
+	cell.Invoke(registerController),
+)
+
+type gatewayApiConfig struct {
+	EnableGatewayAPISecretsSync bool
+	GatewayAPISecretsNamespace  string
+}
+
+func (r gatewayApiConfig) Flags(flags *pflag.FlagSet) {
+	flags.Bool("enable-gateway-api-secrets-sync", r.EnableGatewayAPISecretsSync, "Enables fan-in TLS secrets sync from multiple namespaces to singular namespace (specified by gateway-api-secrets-namespace flag)")
+	flags.String("gateway-api-secrets-namespace", r.GatewayAPISecretsNamespace, "Namespace having tls secrets used by CEC for Gateway API")
+}
+
+func registerController(lc hive.Lifecycle, config gatewayApiConfig) error {
+	if !operatorOption.Config.EnableGatewayAPI {
+		return nil
+	}
+
+	// Setting global logger for controller-runtime
+	ctrlRuntime.SetLogger(logrusr.New(logging.DefaultLogger, logrusr.WithName("controller-runtime")))
+
+	gatewayController, err := NewController(
+		config.EnableGatewayAPISecretsSync,
+		config.GatewayAPISecretsNamespace,
+		operatorOption.Config.ProxyIdleTimeoutSeconds,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create gateway controller: %w", err)
+	}
+
+	lc.Append(hive.Hook{
+		OnStart: func(_ hive.HookContext) error {
+			go gatewayController.Run()
+			return nil
+		},
+	})
```

Comment:
One possible drawback of using a naked goroutine is to have possible false positives when using this cell in tests with goleak. Alternatively, we can rely on a workerpool with a single worker (to be started and stopped in `OnStart` and `OnStop`, respectively) or even better on a `OneShot` Job registered in a group, that also gives the ability to report its status to the health reporter.
But I think it is better to leave this for a follow up PR (we can also do the same for the Ingress cell).

Reply:
yep, the intention of this PR was more to lift & shift it into its own cell. but agree that there's room for improvement. might come up with a follow up. thanks for the pointer.